src/main/groovy/com/getkeepsafe/dexcount/DexFile.groovy

```diff
@@ -55,6 +60,53 @@ class DexFile {
         return [new DexFile(file, false)]
     }
 
+    private static List<DexFile> extractDexFromAar(File file) {
+        // unzip classes.jar from the AAR
+        def zipfile = new ZipFile(file)
+        def entries = Collections.list(zipfile.entries())
+        def jarFile = entries.find { it.name.matches("classes.jar") }
+        def tempClasses = File.createTempFile("classes", ".jar")
+        tempClasses.deleteOnExit()
+
+        def buf = new byte[4096]
+        zipfile.getInputStream(jarFile).withStream { input ->
+            tempClasses.withOutputStream { output ->
+                def read
+                while ((read = input.read(buf)) != -1) {
+                    output.write(buf, 0, read)
+                }
+                output.flush()
+            }
+        }
+        // convert it to DEX format by using the Android dx tool
+        def androidSdkHome = System.getenv("ANDROID_HOME")
+        if (androidSdkHome == null) {
+            throw new Exception("ANDROID_HOME env variable not defined!")
+        }
+        def buildToolsSubDirs = new File(androidSdkHome + File.separatorChar + "build-tools")
+        def dirs = buildToolsSubDirs.listFiles()
+        if (dirs.length == 0) {
+            throw new Exception("No Build Tools found in " + buildToolsSubDirs.absolutePath)
+        }
+        def dxExe = new File(dirs[0].absolutePath + File.separatorChar + "dx")
+        if (!dxExe.exists()) {
+            throw new Exception("dx tool not found at " + dxExe.absolutePath)
+        }
+        // ~/android-sdk/build-tools/23.0.3/dx --dex --output=temp.dex classes.jar
+        def tempDex = File.createTempFile("classes", ".dex")
+        tempDex.deleteOnExit()
+        def sout = new StringBuffer(), serr = new StringBuffer()
```

Comment:
nit: prefer `StringBuilder` over `StringBuffer`, the latter is better basically only for multi-threaded access.